templates/toil-cluster-role-and-policy.yaml

```diff
@@ -0,0 +1,71 @@
+AWSTemplateFormatVersion: 2010-09-09
+Resources:
+  ClusterRole:
+    Type: 'AWS::IAM::Role'
+    Properties:
+      AssumeRolePolicyDocument:
+        Version: 2012-10-17
+        Statement:
+          - Effect: Allow
+            Principal:
+              Service:
+                - ec2.amazonaws.com
+            Action:
+              - 'sts:AssumeRole'
+      Path: /
+      ManagedPolicyArns:
+        - 'arn:aws:iam::563295687221:policy/toil-cluster-instance-policy'
+  ClusterPolicy:
+    Type: AWS::IAM::ManagedPolicy
+    Properties:
+      Description: Minimal permissions for toil instances.
+      ManagedPolicyName: toil-cluster-instance-policy
+      Path: "/"
+      PolicyDocument:
+        Version: '2012-10-17'
+        Statement:
+          - Effect: Allow
+            Action:
+              - ec2:CancelSpotInstanceRequests
+              - ec2:CreateSecurityGroup
+              - ec2:CreateTags
+              - ec2:DeleteSecurityGroup
+              - ec2:DescribeAvailabilityZones
+              - ec2:DescribeImages
+              - ec2:DescribeInstances
+              - ec2:DescribeInstanceStatus
+              - ec2:DescribeKeyPairs
+              - ec2:DescribeSecurityGroups
+              - ec2:DescribeSpotInstanceRequests
+              - ec2:DescribeSpotPriceHistory
+              - ec2:DescribeVolumes
+              - ec2:RequestSpotInstances
+              - ec2:RunInstances
+              - ec2:StartInstances
+              - ec2:StopInstances
+              - ec2:TerminateInstances
+              - iam:PassRole
```

Comment:
do you happen to know how toil is uses this permission?

Reply:
The leader needs this to set the role on the workers